src/streamlink_cli/main.py

```diff
@@ -101,11 +103,13 @@ def create_output():
         elif args.player_http:
             http = create_http_server()
 
+        title = create_title(streamlink.resolve_url(args.url))
```

Comment:
currently the `first plugin` is used for the **title**,
but if you use embedded plugins it might not be the best title

**example**

https://www.cnews.fr/le-direct

it uses **dailymotion** as 2nd plugin

---

here is a patch which **would use the last plugin**,
removes also the rerun of `resolve_url`

https://gist.github.com/back-to/699722cc180769d11579630c3e256e37

Reply:
Respond again if the patch is ready to be applied. A bit confused by the "not quite possible how I wanted it to work" comment.

I'll apply with `git am` when it's ready this time. :)

Reply:
the patch just removes the reusage of `streamlink.resolve_url`

as it was already used here
https://github.com/Hubcapp/streamlink/blob/6a61f70c44ab1804705c3693fa5c50f8a1ec37e1/src/streamlink_cli/main.py#L527
so you don't run the same script twice

**this part is ready**

---

what I wanted is to get an updated `plugin`
but the streamlink session won't update it automatically

but that is **not related to this PR**
as it would requiere some changes to the session part of streamlink